packages/core-database-sequelize/lib/repositories/transactions.js

```diff
@@ -28,9 +28,7 @@ module.exports = class TransactionsRepository extends Repository {
     if (params.senderId) {
       const senderPublicKey = this.__publicKeyfromSenderId(params.senderId)
 
-      if (senderPublicKey) {
-        params.senderPublicKey = senderPublicKey
-      }
+      params.senderPublicKey = senderPublicKey
```

Comment:
In the legacy method, the conditions are disjunctions (OR) instead of all necessary (AND), so we could return an empty result here, without performing any request.